nautobot/extras/jobs.py

```diff
@@ -187,6 +189,23 @@ def before_start(self, task_id, args, kwargs):
             )
             raise RunJobTaskFailed(f"Job {self.job_model} is not enabled to be run!")
 
+        if self.job_model.is_singleton:
+            is_running = cache.get(self.singleton_cache_key)
+            if is_running:
+                self.logger.error(
+                    "Job %s is a singleton and already running.",
+                    self.job_model,
+                    extra={"object": self.job_model, "grouping": "initialization"},
+                )
+                raise RunJobTaskFailed(f"Job '{self.job_model}' is a singleton and already running.")
+            cache_parameters = {
+                "key": self.singleton_cache_key,
+                "value": 1,
+            }
+            if self.job_model.time_limit:
+                cache_parameters["timeout"] = self.job_model.time_limit
+            cache.set(**cache_parameters)
```

Comment:
Sorry this might be a stupid question, but should this be:
```suggestion
            cache_parameters = {
                self.singleton_cache_key: {
                     "key": self.singleton_cache_key,
                     "value": 1,
                }
            }
            if self.job_model.time_limit:
                cache_parameters[self.singleton_cache_key]["timeout"] = self.job_model.time_limit
            cache.set(**cache_parameters)
```
Since we are trying to check if `self.singleton_cache_key` is in the cache or not on line 193?

Reply:
Never mind, checked manually in `nbshell` and it works as expected.